OpenOversight/app/commands.py

```diff
@@ -411,6 +421,388 @@ def bulk_add_officers(filename, no_create, update_by_name, update_static_fields)
         return len(ImportLog.created_officers), len(ImportLog.updated_officers)
 
 
+def _create_or_update_model(
+    row, existing_model_lookup, create_method, update_method, force_create=False, model=None
+):
+    if not row["id"]:
+        return create_method(row)
+    else:
+        if not force_create:
+            return update_method(row, existing_model_lookup[int(row["id"])])
+        else:
+            if model is not None:
+                existing = model.query.filter_by(id=int(row["id"])).first()
```

Comment:
Could an `upsert` be better here than using the ORM and going one-by-one?

Reply:
Good point. Although the way I implemented it, there is just a little bit too much logic happening to make it work I think.

In the normal case, if no id field is provided it has to be an insert, and if the id field is provided it needs to be an update, the imported csv should not have the power to decide the id. With the `--force-create` it might make sense, although I think there is a difference to entries related via foreign keys between deleting an entry and creating a new one with the same id and just updating the entry